### ISSUE TITLE
Can we offload syntax highlighting to bat or other external tools?

### ISSUE THREAD

**Issue by cloud-on-prem** [kyoheiu/felix#244]
Looks like syntax highlighting was removed.
Wondering if we can offload the syntax highlighting part to bat or other external tools?

**Comment by kyoheiu** [comment]
Actually some time ago I'd tried to integrate bat to introduce syntax highlighting, and it did not work as I correctly remember. But things might change, so I'll look at that one more time. Thank you for the suggestion.

**Comment by kyoheiu** [comment]
I found the hard part is the same as before: In horizontally split preview area, bat  might be work as intended, but not in vertically split preview (you can switch between them by `s`). because bat does not provide the option to adjust the starting position of the output.

**Comment by balroggg** [comment]
@kyoheiu Is there something wrong with the `syntect` solution from earlier? For me it works fine.

**Comment by kyoheiu** [comment]
The biggest reason to drop it is the maintainability: IMO,having a syntax highlighting was not the role of a simple file manager like this, and the code was bloated by the feature.
As relatively minor issue, `syntect` cannot properly handle the wide char such as CJK for now, which is kind of irritating but not critical though.

**Comment by balroggg** [comment]
Maybe they fixed this by [CHANGELOG.md](https://github.com/trishume/syntect/blob/95df8850c9859c5403408cf8e7b5a2d7cc6cda2b/CHANGELOG.md?plain=1#L5-L7)
but I'm not sure.

**Comment by kyoheiu** [comment]
That was another problem of `syntect`, and I fixed it by this PR https://github.com/trishume/syntect/pull/456 .
But still, `syntect` cannot handle the width of wide chars properly, so texts containing them are wrapped before reaching the end of the line.

**Comment by rafo** [comment]
@kyoheiu @cloud-on-prem 

Maybe its naïve, but does the terminal-width and offset option of bat:


```
      --terminal-width <width>
          Explicitly set the width of the terminal instead of determining it automatically. If
          prefixed with '+' or '-', the value will be treated as an offset to the actual terminal
          width. See also: '--wrap'.
```

not exactly what is needed?

**Comment by kyoheiu** [comment]
@rafo Thank you for the suggestion...

Wow. I've done it:
![screenshot](https://github.com/kyoheiu/felix/assets/61998590/6edceb17-dd7c-4b69-a0cf-d5eae2459385)
I use `bat <file-path> -fpP --wrap character --terminal-width {preview_space_x} -r {preview_space_y}`.
I'm debugging, but it seems to way smoothly work.

**Comment by kyoheiu** [comment]
Created PR #250

**Comment by rafo** [comment]
> Thank you for the suggestion...
 
I'm happy to help! Looks good!

**Comment by kyoheiu** [comment]
Merged and published as v2.10.0. Thank you all!
